app/helpers/template_helpers.py

```diff
@@ -130,9 +130,14 @@ def data_layer_context(
             key: value for key in DATA_LAYER_KEYS if (value := cookie_session.get(key))
         }
         context = [*additional_context, schema_context]
+
         if tx_id_context:
             context.append(tx_id_context)
-        return context
+
+        flattened_context: dict[str, str] = {}
+        for d in context:
```

Comment:
Can the context not be created flattened in the first place? Rather than doing this loop?

Reply:
I've reduced all the extra code and it's flat by default now.